catlin/pkg/validator/content_validator.go

```diff
@@ -51,10 +60,27 @@ func (v *ContentValidator) Validate() Result {
 	}
 
 	annotations := res.GetAnnotations()
+
 	if _, ok := annotations[consts.MinPipelineVersionAnnotation]; !ok {
 		result.Error("%s is missing a mandatory annotation for minimum pipeline version(%q)", resName, consts.MinPipelineVersionAnnotation)
 	}
 
+	if _, ok := annotations[consts.CategoryAnnotation]; !ok {
+		result.Error("%s is missing a mandatory annotation for category(%q)", resName, consts.CategoryAnnotation)
+	} else {
+		resourceCategories := strings.Split(annotations[consts.CategoryAnnotation], ",")
+		for i := range resourceCategories {
+			resourceCategories[i] = strings.TrimSpace(resourceCategories[i])
+		}
+
+		for i := range resourceCategories {
```

Comment:
we can merge the two for loops by using `strings.TrimSpace(resourceCategories[i]` in second for loop